### ISSUE TITLE
Installation problems encountered in development

### ISSUE THREAD

**Issue by liuchengzhi314159** [seemoo-lab/nexmon_csi#359]
I had to use a linux kernel 6.1 system during the installation of the csi extension. Because the /lib/module/5.10.21/bulid directory wasn't there in the 5.10 system.
The official guide says it's already supported up to 5.15. but I see the driver is already supported up to 6.6.
However the code I need for the csi extension finds only 5.10 and the driver is in the file. I would like to know if there is any difference between these drivers, i.e. the so-called csi extension can only be supported by the driver of the particular system supported by the extension, or is the extension independent of the particular driver version, in other words, can I implement the installation of the csi extension on a 6.1 driver.
If it is possible to do a csi extension implementation on 6.1, what should I do? The makefiles for the two implementations are a bit hard to read, it feels like the two files are too different.
If there is no way to implement it, is there any step by step guide to tell me how I can pull it up in /lib/module/5.10/bulid now!

**Comment by jlinktu** [comment]
https://github.com/seemoo-lab/nexmon_csi/issues/331#issuecomment-1866194237

**Comment by liuchengzhi314159** [comment]
Excuse me, 

I need to install nexmon_csi, (aarch64 architecture, linux kernel 6.1) on a raspberry pi 4b. If I understand your answer above correctly, I need to copy the nexmon_csi folder to the /patches/bcm4.455c0/7_45_189 directory, and then copy the 6.1 driver folder from the drivers folder in the /patches/bcm4.455c0/7_45_189 directory to the nexmon_csi folder, and then change all 5.10 to 6.1 in the Makeflie file.

At the same time I have a question, because the soc architecture in the makefile in this folder is armv7, do I just add an if branch, aarch64 architecture case? And then what do I do to fix the python2 problem? Modify the makefile?
















在 2024-09-24 22:04:23，"Jakob Link" ***@***.***> 写道：

#331 (comment)

**Comment by jlinktu** [comment]
Yes, that describes one possible way how to do it. Go ahead and try it out.

Regarding python2, just make sure that python in your nexmon environment is python2 not python3. There are many ways to achieve that, please consult the web search engine of your choice.

**Comment by liuchengzhi314159** [comment]
Yes, I ran through the installation process at 6pm BST today. The only problem I'm having is that I'm getting a bunch of characters out of make install-firmware, which says that there are a bunch of undefined variables in the makefile in linux-headers-6.1.21+v8, but no errors throughout, so I'm afraid that there's some kind of hidden problem with the process (I'm too good at computers for that). I'm afraid that there are some hidden problems in this process (I'm too computer illiterate to make a judgement on this), and I may need some time to judge them. I may need some time to judge this.

I need to record the timestamps of csi udp to linux kernel for secondary development so that I can do further research (required for postgraduate studies).
















At 2024-09-25 19:16:17, "Jakob Link" ***@***.***> wrote:

Yes, that describes one possible way how to do it. Go ahead and try it out.

Regarding python2, just make sure that python in your nexmon environment is python2 not python3. There are many ways to achieve that, please consult the web search engine of your choice.

**Comment by jlinktu** [comment]
See #395
